examples/reinforcement_learning/dqn_cartpole.py

```diff
@@ -0,0 +1,187 @@
+    parser.add_argument('--target-type', type=str, default='dqn',
+                        help='Target type', choices=['dqn', 'double_dqn'])
+    parser.add_argument('--reward-scale', type=float, default=1e-2,
+                        help='Reward scale factor')
+    parser.add_argument('--replay-start-size', type=int, default=500,
+                        help='Number of steps after which replay is started')
+    parser.add_argument('--steps-to-decay-epsilon', type=int, default=5000,
+                        help='Number of steps used to linearly decay epsilon')
+    parser.add_argument('--min-epsilon', type=float, default=0.01,
+                        help='Minimum value of epsilon')
+    parser.add_argument('--target-update-freq', type=int, default=100,
+                        help='Frequency of target network update')
+    parser.add_argument('--record', action='store_true', default=True,
+                        help='Record performance')
+    parser.add_argument('--no-record', action='store_false', dest='record')
+    args = parser.parse_args()
+
+    # Initialize an environment
+    env = gym.make(args.env)
+    assert isinstance(env.observation_space, gym.spaces.Box)
+    assert isinstance(env.action_space, gym.spaces.Discrete)
+    ndim_obs = env.observation_space.low.size
+    n_actions = env.action_space.n
+    if args.record:
+        env.monitor.start(args.out, force=True)
+
+    # Initialize variables
+    D = collections.deque(maxlen=10 ** 6)
+    Rs = collections.deque(maxlen=100)
+    step = 0
+
+    # Initialize a model and its optimizer
+    Q = QFunction(ndim_obs, n_actions, n_units=args.unit)
+    if args.gpu >= 0:
+        chainer.cuda.get_device(args.gpu).use()
+        Q.to_gpu(args.gpu)
+    target_Q = copy.deepcopy(Q)
+    opt = optimizers.Adam(eps=1e-2)
+    opt.setup(Q)
+
+    for episode in range(args.episodes):
+
+        obs = env.reset()
+        done = False
+        R = 0.0
+        t = 0
+
+        while not done and t < env.spec.timestep_limit:
+
+            # Epsilon is linearly decayed
+            epsilon = 1.0 if len(D) < args.replay_start_size else \
+                max(args.min_epsilon,
+                    np.interp(
+                        step,
+                        [0, args.steps_to_decay_epsilon],
+                        [1.0, args.min_epsilon]))
+
+            # Select an action epsilon-greedily
+            if np.random.rand() < epsilon:
+                a = env.action_space.sample()
+            else:
+                a = get_greedy_action(Q, obs)
+
+            # Execute an action
+            new_obs, r, done, _ = env.step(a)
+            R += r
+
+            # Store a transition
+            D.append((obs, a, r * args.reward_scale, done, new_obs))
+            obs = new_obs
+
+            # Sample a random minibatch of transitions and replay
+            if len(D) >= args.replay_start_size:
+                samples = random.sample(D, args.batchsize)
```

Comment:
Ditto. random.sample requires "Population must be a sequence or set". Can you use a list for D instead of deque? If it's important to limit the size of the transition containers, then please cast here D to a list or set.